src/propagators.jl

```diff
@@ -0,0 +1,325 @@
+using QuBase
+using Docile
+
+import Base: start, next, done
+
+abstract QuantumPropagator{B<:AbstractBasis,T,A}
+
+type StepPropagator{B<:AbstractBasis,T,A} <: QuantumPropagator{B,T,A}
+    hamiltonian::QuMatrix{B,T,A}
+    init_state::QuVector
+    tlist::Array
+    StepPropagator(hamiltonian,init_state,tlist) = new(hamiltonian,init_state,tlist)
+end
+
+
+@doc """
+Euler Problem
+`hamiltonian` : Hamiltonian of the system
+`init_state`  : initial state of the syste
+`tlist`       : Time step array
+""" ->
+immutable QuEuler{B<:AbstractBasis,T,A} <: QuantumPropagator{B,T,A}
+    hamiltonian::QuMatrix{B,T,A}
+    init_state::QuVector
+    tlist::Array
+    QuEuler(hamiltonian, init_state, tlist) = new(hamiltonian, init_state, tlist)
+end
+
+
+QuEuler{B<:AbstractBasis,T,A}(hamiltonian::QuMatrix{B,T,A}, init_state::QuVector, tlist::Array)=QuEuler{B,T,A}(hamiltonian, init_state, tlist)
```

Comment:
Same here regarding the type of `tlist`. We should also have a an additional argument `options=Dict()` or so, which we will need for the other propagators.